src/main/groovy/edu/oregonstate/mist/inventory/resources/InventoryResource.groovy

```diff
@@ -0,0 +1,168 @@
+package edu.oregonstate.mist.inventory.resources
+
+import com.codahale.metrics.annotation.Timed
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.api.jsonapi.ResourceObject
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+import edu.oregonstate.mist.inventory.core.Inventory
+import edu.oregonstate.mist.inventory.db.InventoryDAO
+import groovy.transform.TypeChecked
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import javax.annotation.security.PermitAll
+import javax.ws.rs.DELETE
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.UriBuilder
+
+@Path("inventory")
+@Produces(MediaType.APPLICATION_JSON)
+@PermitAll
+@TypeChecked
+class InventoryResource extends Resource {
+
+    Logger logger = LoggerFactory.getLogger(InventoryResource.class)
+    UriBuilder builder = UriBuilder.fromUri(endpointUri).path(this.class).path("{id}")
+
+    private InventoryDAO inventoryDAO
+    private URI endpointUri
+
+    private final String QUERY_DB_TYPE = "Query"
+    private final String PROVIDED_DATA_DB_TYPE = "Provided Data"
+
+    InventoryResource(InventoryDAO inventoryDAO, URI endpointUri) {
+        this.inventoryDAO = inventoryDAO
+        this.endpointUri = endpointUri
+    }
+
+    /**
+     * Get all inventory objects that aren't deleted
+     * @return
+     */
+    @Timed
+    @GET
+    Response getAllInventories() {
+        List<Inventory> inventories = inventoryDAO.getInventories()
+        List<ResourceObject> resourceObjects = getResourceObjects(inventories)
+
+        ok(new ResultObject(data: resourceObjects)).build()
+    }
+
+    /**
+     * Get one inventory object by ID
+     * @param inventoryID
+     * @return
+     */
+    @Timed
+    @GET
+    @Path('{id: [0-9a-zA-Z-]+}')
+    Response getInventoryByID(@PathParam("id") String inventoryID) {
+        Inventory inventory = inventoryDAO.getInventoryByID(inventoryID)
+
+        if (!inventory) {
+            return notFound().build()
+        }
+
+        ResourceObject resourceObject = getResourceObject(inventory)
+
+        ok(new ResultObject(data: resourceObject)).build()
+    }
+
+    /**
+     * Delete one inventory object by ID
+     * @param inventoryID
+     * @return
+     */
+    @Timed
+    @DELETE
+    @Path('{id: [0-9a-zA-Z-]+}')
+    Response deleteInventoryByID(@PathParam("id") String inventoryID) {
+        Inventory inventory = inventoryDAO.getInventoryByID(inventoryID)
+        Response response
+
+        if (!inventory) {
+            response = notFound().build()
+        } else {
+            try {
```

Comment:
Hmm, we should probably start a transaction at the beginning of this method or a transaction that wraps around the call to deleteInventory(). My thinking is that if deleting some records fail, we don't end up with a half valid record.